### PR TITLE
Cheat Sheet for Google Analytics Regular Expressions

### PR DESCRIPTION
**Pull request by karanchahal**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`Google Analytics Regular Expressions`**
    

###### Description of changes

This PR includes a google-analytics.json file in the \share\goodie\cheat_sheets\json\ folder .
This file is meant to introduce the new cheat sheet.

###### Related Issues or Discussions 

Please let me know what you think

---

Instant Answer Page: https://duck.co/ia/view/google_analytics_regex_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@karanchahal Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by karanchahal** [comment]
I have mentioned the link to my IA page now !

File: share/goodie/cheat_sheets/json/google-analytics-regex.json [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    "name": "GoogleAnalyticsRegex",
+    "description": "Google Analytics Regular Expressions",
+
+    "metadata": {
+        "sourceName": "Cheatography",
+        "sourceUrl" : "https://www.cheatography.com/jay-taylor/cheat-sheets/google-analytics-regular-expressions/"
+    },
+
+    "aliases": [
+        "google analytics regular expressions", "regex google analytics"
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/google-analytics-regex.json]: Nit: Let's have 1 alias per line.

File: share/goodie/cheat_sheets/json/google-analytics-regex.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
+{
+    "id": "google_analytics_regex_cheat_sheet",
+    "name": "GoogleAnalyticsRegex",
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/google-analytics-regex.json]: Nit: Add spacing between the words

**Comment by ManrajGrover** [comment]
@karanchahal Please create IA page named `Google Analytics Regex Cheat Sheet`. This will help Travis identify your IA page and suppress the warning. I've reviewed and asked for few changes. Please fix them for another round of review. 😄

**Comment by karanchahal** [comment]
The new IA page's link is 
https://duck.co/ia/view/google_analytics_regex_cheat_sheet

Please review it and suggest the needed changes :D

**Comment by ManrajGrover** [comment]
@karanchahal Cheat Sheet looks good to me! One thing I can see is that example queries mentioned in IA page are not comma separated. Please make the change. Also, make sure they trigger the IA.

/cc @moollaza

**Comment by karanchahal** [comment]
@ManrajGrover I have changed the IA page according to your requirements and tested the cheatsheet on the beta page successfully.
I'm sorry for the late reply,I was out of the city.
Tell me if anything else is needed.
Thanks !

**Comment by ManrajGrover** [comment]
@karanchahal The issue doesn't look resolved. I see that the page is treating comma separated values as one whole query. Please look into it.

**Comment by sahildua2305** [comment]
@karanchahal I am not sure what's causing the mentioned issue but I will safely blame the comma without trailing space character (see screenshot).

More importantly, you need to mention `cheat sheet` at the end of all the example queries. Since it's a cheat sheet it will be triggered only when the query has `cheat sheet` in it or when it has some other trigger words (depending upon the type of cheatsheet).

google_analytics_regex_cheat_sheet

**Comment by karanchahal** [comment]
Thank you. I fixed the issue.My bad.

File: share/goodie/cheat_sheets/json/google-analytics-regex.json [hunk 3]
```diff
@@ -0,0 +1,162 @@
+{
+    "id": "google_analytics_regex_cheat_sheet",
+    "name": "Google Analytics Regex",
+    "description": "Google Analytics Regular Expressions",
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/google-analytics-regex.json]: @karanchahal can you please change the description to something like 'A list of Regular Expression functionality supported by Google Analytics'  which will be more explicit?

**Comment by karanchahal** [comment]
Anything else ?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Google Analytics Regex cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/google_analytics_regex_cheat_sheet_mobile.png?YMiGMhOA2W)

**Comment by karanchahal** [comment]
Don't know why DaxTheDuck was not deploying to beta before, I committed again with minor changes and it deployed.

**Comment by ManrajGrover** [comment]
@karanchahal No issues with that. This happens sometime. You can ping one of the higher level contributors, if need be and they'll restart the deployment. I feel this can be merged now.

/cc @moollaza

**Comment by daxtheduck** [comment]
@karanchahal this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by sahildua2305** [comment]
Thanks @karanchahal 🍾

**Pull request merged**